pipelines/internal/commands/run/run.go

```diff
@@ -180,30 +181,55 @@ func Invoke(ctx context.Context, service *genomics.Service, project string, argu
 	}
 	fmt.Printf("%s\n", encoded)
 
+	if *attempts < 1 {
+		return fmt.Errorf("The number of attempts must be > 1")
+	}
+
 	if *dryRun {
 		return nil
 	}
 
-	lro, err := service.Pipelines.Run(req).Context(ctx).Do()
-	if err != nil {
-		if err, ok := err.(*googleapi.Error); ok {
-			return fmt.Errorf("starting pipeline: %q: %q", err.Message, err.Body)
+	attempt:=uint(1)
+	for {
+		lro, err := service.Pipelines.Run(req).Context(ctx).Do()
+		if err != nil {
+			if err, ok := err.(*googleapi.Error); ok {
+				return fmt.Errorf("starting pipeline: %q: %q", err.Message, err.Body)
+			}
+			return fmt.Errorf("starting pipeline: %v", err)
 		}
-		return fmt.Errorf("starting pipeline: %v", err)
-	}
 
-	cancelOnInterruptOrTimeout(ctx, service, lro.Name, *timeout)
+		cancelOnInterruptOrTimeout(ctx, service, lro.Name, *timeout)
 
-	fmt.Printf("Pipeline running as %q\n", lro.Name)
-	if *output != "" {
-		fmt.Printf("Output will be written to %q\n", *output)
-	}
+		fmt.Printf("Pipeline running as %q\n", lro.Name)
+		if *output != "" {
+			fmt.Printf("Output will be written to %q\n", *output)
+		}
 
-	if !*wait {
-		return nil
-	}
+		if !*wait {
+			return nil
+		}
 
-	return watch.Invoke(ctx, service, project, []string{lro.Name})
+		if err := watch.Invoke(ctx, service, project, []string{lro.Name}); err != nil {
+			if e, ok := err.(common.PipelineExecutionError); ok && e.IsRetriable() {
+				// non-fatal error
+				if(attempt >= *attempts) {
```

Comment:
This can be tidied up so that you have

if (attempt < *attempts) {
  fmt.Printf("Execution failed: %v (will retry, attempt %d)", err, attempt)
  continue
}
return err